gradio/routes.py

```diff
@@ -444,13 +448,28 @@ def custom_component_path(id: str, type: str, file_name: str):
             if module_path is None or component_instance is None:
                 raise HTTPException(status_code=404, detail="Component not found.")
 
-            return FileResponse(
-                safe_join(
-                    str(Path(module_path).parent),
-                    f"{component_instance.__class__.TEMPLATE_DIR}/{type}/{file_name}",
-                )
+            path = safe_join(
+                str(Path(module_path).parent),
+                f"{component_instance.__class__.TEMPLATE_DIR}/{type}/{file_name}",
             )
 
+            key = f"{id}-{type}-{file_name}"
+
+            if key not in app.custom_component_hashes:
+                app.custom_component_hashes[key] = hashlib.md5(
```

Comment:
We only need to compute the hash once as changing the frontend files is not supported after the server starts. This route is not used for dev mode.